SPECS/kmod-drbd.spec

```diff
@@ -98,86 +54,98 @@ for the DRBD core and various transports.
 
-for flavor in %flavors_to_build ; do
-    make -C drbd install KDIR=%{kernel_source $flavor} \
+make -C drbd install KDIR=/lib/modules/%{kernel_version}/build \
 	%{?_ofed_version:BUILD_OFED=1} \
 	%{?ofed_kernel_dir:OFED_KERNEL_DIR=%{ofed_kernel_dir}} \
 	%{?_ofed_version:OFED_VERSION=%{_ofed_version}} \
 	%{?with_gcov:GCOV_PROFILE=y} \
 	cmd_depmod=:
-    kernelrelease=$(cat %{kernel_source $flavor}/include/config/kernel.release || make -s -C %{kernel_source $flavor} kernelrelease)
+    kernelrelease=$(cat /lib/modules/%{kernel_version}/build/include/config/kernel.release || make -s -C /lib/modules/%{kernel_version}/build kernelrelease)
     mv drbd/build-current/.kernel.config.gz drbd/k-config-$kernelrelease.gz
-done
 
-%if %{defined suse_kernel_module_package}
-# On SUSE, putting the modules into the default path determined by
-# %kernel_module_package_moddir is enough to give them priority over
-# shipped modules.
-rm -f drbd.conf
-%else
-
-mkdir -p $RPM_BUILD_ROOT/etc/depmod.d
-find $RPM_BUILD_ROOT/lib/modules/*/ -name "*.ko"  -printf "%%P\n" |
+mkdir -p %{buildroot}/etc/depmod.d
+find %{buildroot}/lib/modules/*/ -name "*.ko"  -printf "%%P\n" |
 sort | sed -ne 's,^extra/\(.*\)/\([^/]*\)\.ko$,\2 \1,p' |
 xargs -r printf "override %%-16s * weak-updates/%%s\n" \
-> $RPM_BUILD_ROOT/etc/depmod.d/drbd.conf
-install -D misc/SECURE-BOOT-KEY-linbit.com.der $RPM_BUILD_ROOT/etc/pki/linbit/SECURE-BOOT-KEY-linbit.com.der
-%endif
+> %{buildroot}/etc/depmod.d/drbd.conf
+install -D misc/SECURE-BOOT-KEY-linbit.com.der %{buildroot}/etc/pki/linbit/SECURE-BOOT-KEY-linbit.com.der
+
+%post
+if [ -e "/boot/System.map-%{kernel_version}" ]; then
+    /usr/sbin/depmod -aeF "/boot/System.map-%{kernel_version}" "%{kernel_version}" > /dev/null || :
+fi
+
+modules=( $(find /lib/modules/%{kernel_version}/extra/drbd | grep '\.ko$') )
+if [ -x "/sbin/weak-modules" ]; then
+    printf '%s\n' "${modules[@]}"     | /sbin/weak-modules --add-modules
+fi
+
+%preun
+rpm -ql kmod-drbd-%{version}_%{kernel_version}-1.$(arch) | grep '\.ko$' > /var/run/rpm-kmod-drbd-modules
+
+%postun
+if [ -e "/boot/System.map-%{kernel_version}" ]; then
+    /usr/sbin/depmod -aeF "/boot/System.map-%{kernel_version}" "%{kernel_version}" > /dev/null || :
+fi
+
+modules=( $(cat /var/run/rpm-kmod-drbd-modules) )
+rm /var/run/rpm-kmod-drbd-modules
+if [ -x "/sbin/weak-modules" ]; then
+    printf '%s\n' "${modules[@]}"     | /sbin/weak-modules --remove-modules
+fi
+
+%files
+%defattr(-,root,root)
+%doc COPYING
+%doc ChangeLog
+%doc drbd/k-config-*.gz
+%defattr(644,root,root,755)
+/etc/depmod.d/drbd.conf
+/etc/pki/linbit/SECURE-BOOT-KEY-linbit.com.der
+/lib/modules/%{kernel_version}/extra/drbd/*.ko
+/lib/modules/%{kernel_version}/extra/drbd/drbd-kernel-compat/handshake/*.ko
 
 %clean
 rm -rf %{buildroot}
 
 %changelog
+* Mon Jul 15 2024 Ronan Abhamon <ronan.abhamon@vates.tech> - 9.2.10-1.0.xcpng.1
+- Add 0001-Revert-drbd-rework-autopromote.patch
+
```

Comment:
It's a bit more than that. Might be worth splitting this commit. A first commit to fix the spec file for xcpng builds. A second one for the autopromote revert patch.